packages/block/CHANGELOG.md

```diff
@@ -6,6 +6,116 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 (modification: no type change headlines) and this project adheres to
 [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [3.0.0] - 2020-04-01
+
+Verion 3.0.0 brings modernizations to the code and some **breaking changes** you should be aware of.
+
+### TypeScript/Library Import
+
+First TypeScript based release of the library. The import structure has slightly changed along:
+
+**TypeScript**
+
+```typescript
+import { BlockHeader } from 'ethereumjs-block'
+import { Block } from 'ethereumjs-block'
+```
+
+**JavaScript/Node.js**
+
+```javascript
+const BlockHeader = require('ethereumjs-block').BlockHeader
+const Block = require('ethereumjs-block').Block
+```
+
+The library now also comes with a **type declaration file** distributed
+along with the package published.
+
+### Promise-based API
+
+The API of this library is now completely promise-based, the old callback-style
+interface has been dropped.
+
+This affects the following methods of the API now being defined as `async` and
+returning a `Promise`:
+
+**Header Class**
+
+- `BlockHeader.validate(blockchain: Blockchain, height?: BN): Promise<void>`
+
+**Block Class**
+
+- `Block.genTxTrie(): Promise<void>`
+- `Block.validate(blockChain: Blockchain): Promise<void>`
+- `Block.validateUncles(blockchain: Blockchain): Promise<void>`
+
+Usage example:
+
+```javascript
+try {
+  await block.validate(blockchain)
+  // Block validation has passed
+} catch (err) {
+  // handle errors appropriately
+}
+```
+
+### Different signature for constructor
+
+From now on, it's not allowed to initialize `Block` with a `null` value. If for any reason you need to initialize it without defining a value, use the more semantic `undefined` like the examples below:
+
+```typescript
+const b = new Block(undefined, options)
+```
+
+or just:
+
+```typescript
+const b = new Block()
+```
+
+### Change Summary
+
+Other changes along with the `TypeScript` transition PR
+[#72](https://github.com/ethereumjs/ethereumjs-block/pull/72)
+
+- Added Node `10`, `12` support, dropped Node `7` support
+- Browser test run on CI
+- Karma browser test run config modernization and simplification
+- Removal of the `async` dependency
+- Update `ethereumjs-common` dependency from `v1.1.0` to `v1.5.0`
+- Update `ethereumjs-tx` dependency from `v1.2.2` to `v2.1.1`
+- Update `ethereumjs-util` dependency from `v5.0.0` to `v6.1.0`
+- Updated test source files to `TypeScript`
+- Signature fix for pre-homestead blocks, see
+  [#67](https://github.com/ethereumjs/ethereumjs-block/issues/67)
+
+[3.0.0]: https://github.com/ethereumjs/ethereumjs-vm/compare/%40ethereumjs%2Fblock%402.2.0...%40ethereumjs%2Fblock%403.0.0
+
+## [2.2.2] - 2019-12-17
```

Comment:
picked 3.0.0, 2.2.1 and 2.2.2 changelog entries from its own branches in ethereumjs-block. Modified a bit 3.0.0 text.

Reply:
looks great 👍